src/libraries/Pool.sol

```diff
@@ -381,21 +385,17 @@ library Pool {
                 state.amountCalculated = state.amountCalculated - (step.amountIn + step.feeAmount).toInt256();
             }
 
-            // if the protocol fee is on, calculate how much is owed, decrement feeAmount, and increment protocolFee
             if (cache.protocolFee > 0) {
-                // calculate the amount of the fee that should go to the protocol
-                uint256 delta = step.feeAmount * cache.protocolFee / ProtocolFeeLibrary.BIPS_DENOMINATOR;
-                // subtract it from the regular fee and add it to the protocol fee
                 unchecked {
-                    step.feeAmount -= delta;
-                    feeForProtocol += delta;
+                    feeForProtocol += FullMath.mulDiv(step.feeAmount, cache.protocolFee, swapFee + cache.protocolFee);
```

Comment:
I think we can actually leave this block pretty similar to before 
```
// calculate the amount of the fee that should go to the protocol
uint256 delta = FullMath.mulDiv(...)
// subtract it from the regular fee and add it to the protocol fee
unchecked {
       step.feeAmount -= delta;
       feeForProtocol += delta;
}
```

That way we then just have the swap fee left in step.feeAmount and the block below can be left the same too! The protocol fee and swap fee have to add to the total step.feeAmount so no need to calculate them both separately :)